### PR TITLE
Add support for ESP32-WROVER-DEV boards with Camera

### PR DESCRIPTION
**Pull request by gitcnd**
This adds support for the ESP32-WROVER-DEV boards (alternatively named ESP32-WROVER-CAM by other manufacturers)

```
Board Feature			Circuitpython Status
=============                   ====================
Take photos with camera		Tested + Working
Wifi				Tested + Working
Inbuilt RED LED                 Tested + Working
ESP32-CAM-MB Button		Tested + Working
Bluetooth			Not tested
Others				There are no other features I know of
```

Here is my working sample script which takes a photo (and saves it to "out.jpg" in flash)

```python
import io
import os
import board
import espcamera
import socketpool

cam = espcamera.Camera(
    data_pins=board.CAMERA_DATA,
    external_clock_pin=board.CAMERA_XCLK,
    pixel_clock_pin=board.CAMERA_PCLK,
    vsync_pin=board.CAMERA_VSYNC,
    href_pin=board.CAMERA_HREF,
    powerdown_pin=board.CAMERA_PWDN,
    reset_pin=None,
    i2c=board.I2C(),
    external_clock_frequency=20_000_000,
    pixel_format=espcamera.PixelFormat.JPEG,
    frame_size=espcamera.FrameSize.SVGA,
    # jpeg_quality=0,
    # framebuffer_count=0,
    # grab_mode=espcamera.GrabMode.WHEN_EMPTYi
    )

if True:
    frame = cam.take(1)
    fn="out.jpg"
    if isinstance(frame, memoryview):
        jpeg = frame
        print(f"Captured {len(jpeg)} bytes of jpeg data")
        with open(fn, 'wb') as f:
            f.write(jpeg)
        print(f"written to {fn}")

```

Here is the button test code

```python
import board
import digitalio
import time

# Set up the button pin
button = digitalio.DigitalInOut(board.BUTTON)
button.direction = digitalio.Direction.INPUT
button.pull = digitalio.Pull.UP  

while True:
    button_state = button.value
    if button_state:
        print("Button is released")
    else:
        print("Button is pressed")
    time.sleep(0.7) 
```

This board uses different camera pins to my previous PR for the ESP32-CAM, however, python code written for either board works on both of these without change.

Here are my notes:-

pic_2024-05-10_03 40 50_562

### CODE REVIEW

**Comment by gitcnd** [comment]
Be aware of a common hardware fault on some of these - the power regulators are backwards ...

You may need to swap the 2 power regulators (65BX and 54FK) which got soldered in the wrong way around! See pic - cheap copy at top, original at bottom :-

a5d0fe8f1b61a2e39d502354380d51127bafeaae

File: ports/espressif/boards/esp32-wrover-dev-cam/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#define MICROPY_HW_BOARD_NAME       "Espressif ESP32-WROVER-DEV-CAM"
```
Comment (tannewt) [inline ports/espressif/boards/esp32-wrover-dev-cam/mpconfigboard.h:27]: This board isn't made by Espressif so let's use the name on the board.

```suggestion
#define MICROPY_HW_BOARD_NAME       "Freenove ESP32-WROVER-DEV-CAM"
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for the board def. Let's not attribute this board to espressif at all. Instead let's add a separate creator id for freenove.

File: ports/espressif/boards/esp32-wrover-dev-cam/mpconfigboard.h (hunk 1 above)
Comment (gitcnd) [reply ports/espressif/boards/esp32-wrover-dev-cam/mpconfigboard.h:27]: Do you know if they made it?
None of my boards say "Freenove" on them, except one that came with a robot dog I bought - the brand of which is Freenove.  As far as I can tell, this is an espressif reference design, manufactured by a bunch of different places?
I've written to Freenove to ask them if they designed this or not - better to get the attribution correct than guess :-)

Comment (tannewt) [reply ports/espressif/boards/esp32-wrover-dev-cam/mpconfigboard.h:27]: Feel free to make an "Unknown" creator id too. I just don't want to misattribute it.

Comment (gitcnd) [reply ports/espressif/boards/esp32-wrover-dev-cam/mpconfigboard.h:27]: Good call @tannewt !  They wrote back.

I'll convert this PR into a draft while I work fix all the ID codes and perform the re-naming of things.


```
Dear customer,

Thank you for choosing Freenove. We are fully responsible for any concerns.

The one with our logo on it is ours, which is the original design.
The ones without logo are counterfeit.

If you have any concerns, please feel free to contact us.

Have a great day!

Best regards,
Shaynee
Freenove Support Team
--------------------------
www.freenove.com 
```

**Comment by gitcnd** [comment]
OK - got a new ID sorted for Freenove, and updated the name and ID in the relevant files.

**Comment by gitcnd** [comment]
Should be ready to merge now - the PR for the new IDs is https://github.com/creationid/creators/pull/69

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**